### PR TITLE
1D tensors as bar charts

### LINKED ISSUE #3709: rr.Tensor not working for 1D tensors
**Describe the bug**
When logging a 1D tensor I'm getting "Expected a 2D slice" in the viewer. Before, this would give a bar chart-like plot.

**To Reproduce**
```python
import rerun as rr
import torch
rr.init("tensor", spawn=True)
rr.log("tensor", rr.Tensor(torch.rand(10,)))
```

**Expected behavior**
Previous behavior seems better.

**Rerun version**
```
rerun_py 0.9.0 [rustc 1.72.1 (d5c2e9c34 2023-09-13), LLVM 16.0.5] x86_64-unknown-linux-gnu 4f56b02, built 2023-10-05T15:37:48Z
```

### PR DESCRIPTION
**Pull request by teh-cmc**
Relax the bar-chart view's requirements, and strengthen the tensor view's requirements:
- make bar-chart look for either `BarChartIndicator` _or_ `TensorIndicator`
- make tensor ignore anything with less than 2 dimensions (it can't display those anyway!)

---

```py
import rerun as rr
import torch
rr.init("tensor", spawn=True)
rr.log("tensor", rr.Tensor(torch.rand(10,)))
```
image

---

- Resolves #3709  

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3769) (if applicable)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/3769)
- [Docs preview](https://rerun.io/preview/dcd03363c86c6242023a10c9715643cf9f601d58/docs) 
- [Examples preview](https://rerun.io/preview/dcd03363c86c6242023a10c9715643cf9f601d58/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

**Review by emilk — APPROVED** [review]

**Comment by nikolausWest** [comment]
I wonder if it would have made more sense to add an extra dimension instead and treat it like a 1 x N tensor? That way we make it possible for the user to choose what way they'd like to see their data. Like a tensor row or a bar chart depending on the indicator
